plugins/clnrest/utilities/generate_certs.py

```diff
@@ -5,98 +5,65 @@
-    for entity_type in ["server", "client"]:
-        public_key = server_public_key if entity_type == "server" else client_public_key
+def generate_cert(entity_type, ca_subject, ca_private_key, rest_host, certs_path):
+    # Generate Key pair
+    private_key = ec.generate_private_key(ec.SECP256R1())
+    public_key = private_key.public_key()
 
+    # Generate Certificates
+    if isinstance(ca_subject, x509.Name):
         subject = x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, f"cln rest {entity_type}")])
-
+        cert_builder = create_cert_builder(subject, ca_subject, public_key, rest_host)
+        cert = cert_builder.sign(ca_private_key, hashes.SHA256())
+    else:
+        ca_subject = x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, u"cln Root REST CA")])
+        ca_private_key, ca_public_key = private_key, public_key
+        cert_builder = create_cert_builder(ca_subject, ca_subject, ca_public_key, rest_host)
         cert = (
-            x509.CertificateBuilder()
-            .subject_name(subject)
-            .issuer_name(ca_subject)
-            .public_key(public_key)
-            .serial_number(x509.random_serial_number())
-            .not_valid_before(datetime.datetime.utcnow())
-            .not_valid_after(datetime.datetime.utcnow() + datetime.timedelta(days=10 * 365))  # Ten years validity
-            .add_extension(x509.SubjectAlternativeName([x509.DNSName(u"cln"), x509.DNSName(u'localhost'), x509.IPAddress(ipaddress.IPv4Address(u'127.0.0.1'))]), critical=False)
+            cert_builder
+            .add_extension(x509.BasicConstraints(ca=True, path_length=None), critical=True)
             .sign(ca_private_key, hashes.SHA256())
         )
 
-        # Serialize Server and Client certificates and write to disk
-        with open(os.path.join(certs_path, f"{entity_type}.pem"), "wb") as f:
-            f.write(cert.public_bytes(serialization.Encoding.PEM))
-
-    # Serialize Private Keys (Server)
-    with open(os.path.join(certs_path, "server-key.pem"), "wb") as f:
-        f.write(server_private_key.private_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PrivateFormat.PKCS8,
-            encryption_algorithm=serialization.NoEncryption()
-        ))
-
-    # Serialize Private Keys (Client)
-    with open(os.path.join(certs_path, "client-key.pem"), "wb") as f:
-        f.write(client_private_key.private_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PrivateFormat.PKCS8,
-            encryption_algorithm=serialization.NoEncryption()
-        ))
+    os.makedirs(certs_path, exist_ok=True)
+    save_cert(entity_type, cert, private_key, certs_path)
+    return ca_subject, ca_private_key
 
 
-def generate_certs(plugin, certs_path):
-    ca_subject, ca_private_key = generate_ca_cert(certs_path)
-    generate_client_server_certs(certs_path, ca_subject, ca_private_key)
+def generate_certs(plugin, rest_host, certs_path):
```

Comment:
Nit: do we want to overwrite the CA and server certs if they exist? It seems like trusting a new CA is kind of a pain if you don't have to do it.  Then again, if the client cert is totally missing, maybe it's the right choice to refresh it all.